### PR TITLE
Switch Dockerfile to python 3

### PR DESCRIPTION
**Pull request by d2dyno**
per blog: "We plan on ending mainstream support on 1st April 2019, where upon Python 3.5+ will be the only officially supported platform"
https://matrix.org/blog/2018/12/21/porting-synapse-to-python-3/

### CODE REVIEW

**Comment by Half-Shot** [comment]
(edited title to reflect that this is a docker PR)

**Comment by d2dyno** [comment]
Thanks!

I tested this on my local installation (https://hub.docker.com/r/d2dyno/synapse-python3), and thus far have only encountered one minor notice:
`/usr/local/lib/python3.7/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25.2) or chardet (3.0.4) doesn't match a supported version!
RequestsDependencyWarning)`

**Comment by anoadragon453** [comment]
@d2dyno We're aware of the urllib3 issue and should be unrelated to this PR specifically.

File: changelog.d/5132.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Docker container now builds with Python 3.7.
```
[the remark this answers is not in the document]
Comment (richvdh) [reply changelog.d/5132.feature]: Suggest making this a `misc` file rather than a `feature`. It's unlikely to affect the vast majority of regular users.

**Comment by hawkowl** [comment]
Fixed by way of https://github.com/matrix-org/synapse/pull/5546, which sets it to be the default to be built and sent to Docker Hub as well.

**Pull request closed without merging**